pyramid_closure/views.py

```diff
@@ -19,15 +16,19 @@ def depsjs(request):
     settings = request.registry.settings
     pyramid_closure = settings.get("pyramid_closure")
 
-    roots = pyramid_closure.get("roots") if pyramid_closure else settings.get("pyramid_closure.roots")
+    roots = pyramid_closure.get("roots") if pyramid_closure else \
+        settings.get("pyramid_closure.roots")
     roots = aslist(roots or [])
-
-    roots_with_prefix = (
-        pyramid_closure.get("roots_with_prefix")
-        if pyramid_closure
-        else settings.get("pyramid_closure.roots_with_prefix")
-    )
-    roots_with_prefix = pairwise(roots_with_prefix)
+    for root in roots:
+        path_to_source.update(depswriter._GetRelativePathToSourceDict(root))
+
+    roots_with_prefix = pyramid_closure.get("roots_with_prefix") if \
+        pyramid_closure else \
+        settings.get("pyramid_closure.roots_with_prefix")
+    roots_with_prefix = aslist(roots_with_prefix or [])
```

Comment:
in c2c.template case the roots_with_prefix will be a dictionary and this line will break it ...
`aslist({'aaa': 222}) => ['aaa']`